File: MultiHotelManagement/src/main/java/com/MHM/MultiHotelManagement/service/InvoiceService.java
package com.MHM.MultiHotelManagement.service;

import com.MHM.MultiHotelManagement.dto.request.InvoiceRequestDTO;
import com.MHM.MultiHotelManagement.dto.response.InvoiceResponseDTO;

import java.util.List;

public interface InvoiceService {
    InvoiceResponseDTO create(InvoiceRequestDTO dto);
    InvoiceResponseDTO getById(Long id);
    List<InvoiceResponseDTO> getByCustomer(Long customerId);
    List<InvoiceResponseDTO> getByBooking(Long bookingId);
    List<InvoiceResponseDTO> getByHotelId(Long hotelId);
    List<InvoiceResponseDTO> getByOwnerId(Long ownerId);
    List<InvoiceResponseDTO> getAll();
    void delete(Long id);
}

File: MultiHotelManagement/frontend/src/app/services/invoice.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Invoice } from '../models';

@Injectable({ providedIn: 'root' })
export class InvoiceService {
  private baseUrl = 'http://localhost:8080/api/invoices';

  constructor(private http: HttpClient) {}

  getAll(): Observable<Invoice[]> {
    return this.http.get<Invoice[]>(this.baseUrl);
  }

  getById(id: number): Observable<Invoice> {
    return this.http.get<Invoice>(`${this.baseUrl}/${id}`);
  }

  getByCustomer(customerId: number): Observable<Invoice[]> {
    return this.http.get<Invoice[]>(`${this.baseUrl}/customer/${customerId}`);
  }

  getByBooking(bookingId: number): Observable<Invoice[]> {
    return this.http.get<Invoice[]>(`${this.baseUrl}/booking/${bookingId}`);
  }

  getByHotelId(hotelId: number): Observable<Invoice[]> {
    return this.http.get<Invoice[]>(`${this.baseUrl}/hotel/${hotelId}`);
  }

  getByOwnerId(ownerId: number): Observable<Invoice[]> {
    return this.http.get<Invoice[]>(`${this.baseUrl}/owner/${ownerId}`);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
